### PR TITLE
Moves scaling out of experimental

### PR DESCRIPTION
**Pull request by elijahbenizzy**
[Short description explaining the high-level reason for the pull request]

## Checklist

- [ ] PR has an informative and human-readable title (this will be pulled into the release notes)
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code passed the pre-commit check & code is left cleaner/nicer than when first encountered.
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future TODOs are captured in comments
- [ ] Project documentation has been updated if adding/changing functionality.

### CODE REVIEW

File: hamilton/registry.py [hunk 1]
```diff
@@ -76,15 +76,7 @@ def load_extension(plugin_module: str):
...
     """
-    mod = importlib.import_module(f"hamilton.plugins.{plugin_module}_extensions")
-    assert hasattr(mod, "register_types"), "Error extension missing function register_types()"
-    assert hasattr(
-        mod, f"get_column_{plugin_module}"
-    ), f"Error extension missing get_column_{plugin_module}"
-    assert hasattr(
-        mod, f"fill_with_scalar_{plugin_module}"
-    ), f"Error extension missing fill_with_scalar_{plugin_module}"
-    logger.info(f"Detected {plugin_module} and successfully loaded Hamilton extensions.")
```
Comment (skrawcz) [inline hamilton/registry.py:87]: can we not make this conditional?
e.g. have the module declare some variable, and if that's not present we do these asserts?

**Comment by skrawcz** [comment]
Need a README somewhere too (maybe in plugins) to explain how to add/adjust and name files.

**Review by skrawcz — COMMENTED** [review]
Plugins will get large. We also don't want very large files either.

So the evolution should be into packages, right?
h_spark.py -->

 - __init__.py 
 -  ... modules organized however we want


And then similarly for *_extensions.py...?

File: hamilton/registry.py (hunk 1 above)
Comment (elijahbenizzy) [reply hamilton/registry.py:87]: Sure, that works

**Comment by elijahbenizzy** [comment]
> Plugins will get large. We also don't want very large files either.
> 
> So the evolution should be into packages, right? h_spark.py -->
> 
> * **init**.py
> * ... modules organized however we want
> 
> And then similarly for *_extensions.py...?

Yeah, I think that'll be the best way for it to evolve. Should be able to do it and keep compatible references later on, no need to break into packages yet.

**Review by skrawcz — APPROVED** [review]
I think things work.

**Pull request merged**